pkg/provider/azure_loadbalancer.go

```
@@ -1933,161 +1857,292 @@ func lbRuleConflictsWithPort(rule network.LoadBalancingRule, frontendIPConfigID
-			// Since the port is always 0 when enabling HA, only one rule should be configured.
-			break
+	// HA mode forward traffic of all port to backend
+	// It is only supported on standard loadbalancer in internal mode
+	if consts.IsK8sServiceUsingInternalLoadBalancer(service) &&
+		az.useStandardLoadBalancer() &&
+		consts.IsK8sServiceHasHAModeEnabled(service) {
+
+		props := &network.LoadBalancingRulePropertiesFormat{
+			Protocol: network.TransportProtocolAll,
+			FrontendIPConfiguration: &network.SubResource{
+				ID: to.StringPtr(lbFrontendIPConfigID),
+			},
+			BackendAddressPool: &network.SubResource{
+				ID: to.StringPtr(lbBackendPoolID),
+			},
+			LoadDistribution:     loadDistribution,
+			FrontendPort:         to.Int32Ptr(0),
+			BackendPort:          to.Int32Ptr(0),
+			DisableOutboundSnat:  to.BoolPtr(az.disableLoadBalancerOutboundSNAT()),
+			EnableFloatingIP:     to.BoolPtr(true),
+			EnableTCPReset:       to.BoolPtr(true),
+			IdleTimeoutInMinutes: lbIdleTimeout,
 		}
 
-		lbRuleName := az.getLoadBalancerRuleName(service, port.Protocol, port.Port)
+		lbRuleName := az.getloadbalancerHAmodeRuleName(service)
 		klog.V(2).Infof("getExpectedLBRules lb name (%s) rule name (%s)", lbName, lbRuleName)
-
-		transportProto, _, probeProto, err := getProtocolsFromKubernetesProtocol(port.Protocol)
-		if err != nil {
-			return expectedProbes, expectedRules, err
+		//we only add one probe rule here for lb rule's reference
+		var probe *network.Probe
+		if nodeEndpointHealthprobe != nil {
+			probe = nodeEndpointHealthprobe
+		} else {
+			err := processDeprecatedHealthProbeAnnotations(service)
+			if err != nil {
+				return expectedProbes, expectedRules, err
+			}
+			for _, port := range service.Spec.Ports {
+				probeProps, err := az.buildHealthProbeRulesForPort(service.Annotations, port)
+				if err != nil {
+					klog.V(2).ErrorS(err, "error occurred when buildHealthProbeRulesForPort", "service", service.Name, "namespace", service.Namespace,
```

Why not return the error here?

---

will return the error instead